src/leveled_sst.erl

```diff
@@ -2560,66 +2424,141 @@ binaryslot_trimmedlist(FullBin, StartKey, EndKey,
+            ++ in_range(deserialise_block(B5, PressMethod), StartKey, EndKey);
+        all_blocks ->
+            get_lefthand_blocks(B1, B2, PressMethod, StartKey)
+            ++ MidBlockList
+            ++ get_righthand_blocks(B4, B5, PressMethod, EndKey);
+        lt_mid ->
+            in_range(
+                get_lefthand_blocks(B1, B2, PressMethod, StartKey),
+                all,
+                EndKey);
+        le_mid ->
+            get_lefthand_blocks(B1, B2, PressMethod, StartKey)
+            ++ in_range(MidBlockList, all, EndKey);
+        mid_only ->
+            in_range(MidBlockList, StartKey, EndKey);
+        ge_mid ->
+            in_range(MidBlockList, StartKey, all)
+            ++ get_righthand_blocks(B4, B5, PressMethod, EndKey);
+        gt_mid ->
+            in_range(
+                get_righthand_blocks(B4, B5, PressMethod, EndKey),
+                StartKey,
+                all)
+    end.
+
+get_lefthand_blocks(B1, B2, PressMethod, StartKey) ->
+    BlockList2 = deserialise_block(B2, PressMethod),
+    case previous_block_required(
+            fetchends_rawblock(BlockList2), StartKey) of
         true ->
-            [MidBlockList];
+            in_range(deserialise_block(B1, PressMethod), StartKey, all)
+            ++ BlockList2;
         false ->
-            [MidBlockList, Block4, Block5]
+            in_range(BlockList2, StartKey, all)
+    end.
+
+get_righthand_blocks(B4, B5, PressMethod, EndKey) ->
+    BlockList4 = deserialise_block(B4, PressMethod),
+    case next_block_required(
+            fetchends_rawblock(BlockList4), EndKey) of
+        true ->
+            BlockList4
+            ++ in_range(deserialise_block(B5, PressMethod), all, EndKey);
+        false ->
+            in_range(BlockList4, all, EndKey)
+    end.
+
+filterby_midblock({not_present, not_present}, _RangeKeys) ->
+    empty;
+filterby_midblock(
+        {_MidFirst, MidLast}, {StartKey, _EndKey}) when StartKey > MidLast ->
+    gt_mid;
+filterby_midblock(
+        {MidFirst, MidLast}, {StartKey, EndKey}) when StartKey >= MidFirst ->
+    case leveled_codec:endkey_passed(EndKey, MidLast) of
+        true ->
+            mid_only;
+        false ->
+            ge_mid
     end;
-filter_blocks_required({MidFirst, MidLast}, {_StartKey, EndKey},
-                [Block1, Block2, MidBlockList, Block4, Block5]) ->
+filterby_midblock({MidFirst, MidLast}, {_StartKey, EndKey}) ->
     AllBefore = leveled_codec:endkey_passed(EndKey, MidFirst),
     NoneAfter = leveled_codec:endkey_passed(EndKey, MidLast),
     case {AllBefore, NoneAfter} of
         {true, true} ->
-            [Block1, Block2];
+            lt_mid;
         {false, true} ->
-            [Block1, Block2, MidBlockList];
+            le_mid;
         {false, false} ->
-            [Block1, Block2, MidBlockList, Block4, Block5]
+            all_blocks
     end.
 
+previous_block_required({not_present, not_present}, _SK) ->
```

Comment:
this clause is strictly spoken not needed

Reply:
The function is not called super-frequently, so I don't think the extra cause is harmful ... save a future reviewer having to double-check the impact of erlang term order